# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/ion/bug1745388.js
# Commit: f0fc28562ec4
# Full Hash: f0fc28562ec4396e455f723837906d60b03f4311
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2021-12-15 04:41:24
# Description:
#   Bug 1745388: Compute dependency info for shape guard removal r=jandem
#   
#   If LICM and GVN are both disabled, `ins->dependency()` is never set and we dereference a null pointer.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D133562
# ==============================================================================

diff -r fa1e33c47711 -r f0fc28562ec4 js/src/jit-test/tests/ion/bug1745388.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/ion/bug1745388.js	Tue Dec 14 17:17:26 2021 +0000
@@ -0,0 +1,3 @@
+// |jit-test| --ion-gvn=off; --ion-licm=off
+
+for (var k = 0; k < 100; k++) {}